### PR TITLE
Ciinabox Metrics Recipe and Spec for AWS CW Metrics framework

### PR DESCRIPTION
**Pull request by toshke**

### CODE REVIEW

**Comment by toshke** [comment]
@aaronwalker @Guslington ciinabox-metrics created as standalone recipe

**Comment by toshke** [comment]
Important change to note - is that instance role running this needs to have cloudwatch: PutMetricData permission

File: recipes/default.rb [hunk 1]
```diff
@@ -18,4 +18,5 @@
   include_recipe 'base2::users'
   include_recipe 'base2::environment'
   include_recipe 'base2::nrpe'
+  include_recipe 'base2::aws_cw_metrics'
```
Comment (aaronwalker) [inline recipes/default.rb]: I think we should conditionally include this recipe based on an attribute which can be overridden in the customers cookbook and the default should be to not include it by default. Right now I want this to be an opt-in

Comment (toshke) [inline recipes/default.rb]: @aaronwalker 
`['base2']['aws']['cw-metrics-enabled']` attribute added to optionally include `aws_cw_metrics` recipe

Comment (toshke) [reply recipes/default.rb]: @aaronwalker 

Updated recipes so it's not in default at all, if metrics are required they can be included manually

```
include_recipe 'base2::aws_cw_metrics
```

**Comment by aaronwalker** [comment]
retest this please

**Pull request merged**